libs/libcommon/include/common/logger_useful.h

```diff
@@ -3,16 +3,83 @@
 /// Macros for convenient usage of Poco logger.
 
 #include <sstream>
+#include <atomic>
 #include <Poco/Logger.h>
+#include <Poco/Message.h>
+#include <Poco/Version.h>
+#include <Core/SettingsCommon.h>
+#include <Common/CurrentThread.h>
 
 #ifndef QUERY_PREVIEW_LENGTH
 #define QUERY_PREVIEW_LENGTH 160
 #endif
 
 using Poco::Logger;
+using Poco::Message;
+using DB::LogsLevel;
+using DB::CurrentThread;
 
 /// Logs a message to a specified logger with that level.
 
+#if defined(POCO_CLICKHOUSE_PATCH)
+
+#define LOG_TRACE(logger, message) do { \
```

Comment:
Let's unify these macros with more preprocessor programming.

Reply:
We want to write *#ifdef* and *#ifndef* for every LOG_TRACE, LOG_DEBUG etc. ?

Reply:
No...
You can unify trace/debug/information... macros.

Reply:
Ok